### ISSUE TITLE
Error when using mocking interfaces with type parameters

### ISSUE THREAD

**Issue by dalewking** [mockative/mockative#55]
Using version 1.2.6

I have this type:
```
interface UseCase<in P, out R> where R : Any? {
    suspend operator fun invoke(param: P): Result<R>
}
```

And I tried to mock an instance of it like this:
```
    val foo = mock(classOf<UseCase<Int, String>>())
```

But the generated code:
```
public fun <in P, out R> mock(@Suppress("UNUSED_PARAMETER") type: KClass<UseCase<P, R>>):
    UseCase<P, R> = UseCaseMock<P, R>()
```

fails to build with an error:
> Variance annotations are only allowed for type parameters of classes and interfaces

**Comment by dalewking** [comment]
Looks like even if I drop the in and out from the interface it just moves the error to the generated Mock class complaining that it is missing the in and out

**Comment by Nillerr** [comment]
Hi @dalewking 

Thank you for your patience. I have addressed this issue in [v1.4.0](https://github.com/mockative/mockative/releases/tag/v1.4.0) which I released this morning.
